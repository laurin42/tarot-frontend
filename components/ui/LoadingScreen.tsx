import React from "react";
import { View, ActivityIndicator, StyleSheet } from "react-native";

export function LoadingScreen(): JSX.Element {
  return (
    <View style={styles.container}>
      <ActivityIndicator size="large" color="#A78BFA" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#1F2937",
  },
});
