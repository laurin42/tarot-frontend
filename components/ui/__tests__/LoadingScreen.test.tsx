import React from "react";
import { render } from "@testing-library/react-native";
import { LoadingScreen } from "../LoadingScreen"; // Adjust path if needed

describe("LoadingScreen Component", () => {
  // The snapshot test inherently checks if the ActivityIndicator is rendered
  // as part of the component's output.
  it("renders correctly and matches snapshot", () => {
    const tree = render(<LoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
    // By matching the snapshot, we implicitly confirm the ActivityIndicator
    // and the surrounding View structure are rendered as expected.
  });
});
