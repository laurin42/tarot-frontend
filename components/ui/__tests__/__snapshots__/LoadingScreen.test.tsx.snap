// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen Component renders correctly and matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1F2937",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#A78BFA"
    size="large"
  />
</View>
`;
